### PR TITLE
Bug 1954892: canary: Add priority class to canary daemonset 

### PR DESCRIPTION
**Pull request by sgreene570**
**canary: Add priority class to canary daemonset**

OpenShift components are required to specify a PriorityClassName for daemonsets/deployments.

This commit ensures that the Canary daemonset has the `system-cluster-critical` priority class set.

assets/canary/daemonset.yaml:
Set the `priorityClassName` field to be `system-cluster-critical`.

pkg/operator/controller/canary/daemonset.go:
Update the Canary daemonset when `priorityClassName` changes.

pkg/operator/controller/canary/daemonset_test.go:
Add a test to ensure that `priorityClassName` is set correctly by `desiredCanaryDaemonSet`.
Add a test to verify that the Canary daemonset is updated when `priorityClassName` is mutated.

This commit is in support of Bug 1954892.

**Update bindata for Canary daemonset changes**

Run `make update` and commit the output.

### CODE REVIEW

File: pkg/operator/controller/canary/daemonset.go [hunk 1]
```diff
@@ -131,6 +131,11 @@ func canaryDaemonSetChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.
 		changed = true
 	}
 
+	if !cmp.Equal(current.Spec.Template.Spec.PriorityClassName, expected.Spec.Template.Spec.PriorityClassName, cmpopts.EquateEmpty()) {
```
Comment (Miciah) [inline pkg/operator/controller/canary/daemonset.go:134]: Any reason to use `cmp.Equal` instead of `==`?

File: assets/canary/daemonset.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ spec:
       annotations:
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
     spec:
+      priorityClassName: openshift-cluster-critical
```
Comment (Miciah) [inline assets/canary/daemonset.yaml:13]: Should that be `system-cluster-critical`?  Where is "`openshift-cluster-critical`" defined?

File: pkg/operator/controller/canary/daemonset.go (hunk 1 above)
Comment (sgreene570) [reply pkg/operator/controller/canary/daemonset.go]: nope. Fixed.

File: assets/canary/daemonset.yaml (hunk 2 above)
Comment (sgreene570) [reply assets/canary/daemonset.yaml]: Yes! Typo :sweat:

**Comment by sgreene570** [comment]
Some operator e2e tests are failing on the following error:
```

**Comment by sgreene570** [comment]
Whoops! Somehow I fat fingered the close and comment button.

**Comment by Miciah** [comment]
Thanks!
/lgtm

**Pull request merged**